File: core/model/src/commonMain/kotlin/io/github/taetae98coding/diary/core/model/routine/Routine.kt
package io.github.taetae98coding.diary.core.model.routine

import kotlin.uuid.Uuid
import kotlinx.datetime.LocalDate

public data class Routine(
    val id: Uuid,
    val detail: RoutineDetail,
    val rRules: List<RoutineRRule>,
    val rDates: Set<LocalDate>,
    val exDates: Set<LocalDate>,
    val isCalendarVisible: Boolean,
    val isFinished: Boolean,
    val isDeleted: Boolean,
    val updatedAt: Long,
    val createdAt: Long,
)

File: supabase/functions/_shared/entity/routine.ts
import { RoutineDetail } from "./routine-detail.ts";
import { RoutineRRule } from "./routine-rrule.ts";

export interface Routine {
  id: string;
  detail: RoutineDetail;
  rRules: RoutineRRule[];
  rDates: string[];
  exDates: string[];
  isCalendarVisible: boolean;
  isFinished: boolean;
  isDeleted: boolean;
  updatedAt: number;
  createdAt: number;
}
